perf(codegen): reduce branches printing `ObjectPattern`

overlookmotel opened the pull request:
Small perf optimization. Make a fast path for printing an empty object literal.

Previously we branched twice on `self.is_empty()` (unpredicable branches), and did an unnecessary call to `print_list` for empty objects. This change should speed up printing both empty objects, and objects which do have properties.

Boshen commented:
### Merge activity

* **Nov 5, 12:09 PM UTC**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc) once it meets the requirements.
* **Nov 5, 12:09 PM UTC**: `Boshen` added this pull request to the [Graphite merge queue](https://app.graphite.com/merges?org=oxc-project&repo=oxc).

overlookmotel committed:
perf(codegen): reduce branches printing `ObjectPattern` (#15316)

Small perf optimization. Make a fast path for printing an empty object literal.

Previously we branched twice on `self.is_empty()` (unpredicable branches), and did an unnecessary call to `print_list` for empty objects. This change should speed up printing both empty objects, and objects which do have properties.
